src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEAFULT_SIZE = 8;
+    private static Object[] stack;
+    private static int size = 0;
 
+    public MateStack() {
+        stack = new Object[DEAFULT_SIZE];
+        Arrays.fill(stack, null);
+    }
+
+    public void push(T value) throws RuntimeException {
```

Comment:
The condition 'size <= stack.length' should be 'size < stack.length'. Otherwise, it will throw a StackOverflowError when the stack is full, but there is still space for one more element.